### PR TITLE
Dont modifiy StatementList when invalid arguments passed to setStatements

### PR DESCRIPTION
**Pull request by Benestar**
@JeroenDeDauw Addresses your comment and adds tests for those cases.

*Edit:* Added a missing `@since 3.0` tag as well.

### CODE REVIEW

**Comment by thiemowmde** [comment]
What is `setStatements`? Who reviewed this? Who needs this and why?

File: tests/unit/Statement/StatementListTest.php [hunk 1]
```diff
@@ -702,4 +703,48 @@ public function testGivenNoTraversable_setStatementsThrowsException() {
 		$statements->setStatements( $statement1 );
 	}
 
+	public function testGivenNoStatement_setStatementsThrowsException() {
```
Comment (JeroenDeDauw) [inline tests/unit/Statement/StatementListTest.php]: Did you mean `FivenNonStatement`?

File: tests/unit/Statement/StatementListTest.php [hunk 2]
```diff
@@ -702,4 +703,48 @@ public function testGivenNoTraversable_setStatementsThrowsException() {
...
+		$statement1 = $this->getStatement( 1, 'guid1' );
+		$statements = new StatementList();
+
+		$this->setExpectedException( 'InvalidArgumentException' );
+		$statements->setStatements( array( $statement1, false ) );
+	}
+
+	public function testGivenNoTraversable_setStatementsDoesNotEditList() {
+		$statement1 = $this->getStatement( 1, 'guid1' );
```
Comment (JeroenDeDauw) [inline tests/unit/Statement/StatementListTest.php]: For this test you do not need all these statements

File: src/Statement/StatementList.php [hunk 3]
```diff
@@ -50,7 +50,7 @@ public function __construct( $statements = array() /*...*/ ) {
 
 	/**
 	 * @since 3.0
-	 *
```
Comment (thiemowmde) [inline src/Statement/StatementList.php]: I think this is why #482 fails.

**Comment by JeroenDeDauw** [comment]
@thiemowmde you can easily find that out yourself

File: tests/unit/Statement/StatementListTest.php (hunk 1 above)
Comment (Benestar) [inline tests/unit/Statement/StatementListTest.php]: Perhaps that is the right way to describe the test in English.

File: tests/unit/Statement/StatementListTest.php (hunk 2 above)
Comment (Benestar) [inline tests/unit/Statement/StatementListTest.php]: One statement is enough, indeed. However, it does not hurt to have more of them. Is this an issue?

File: src/Statement/StatementList.php (hunk 3 above)
Comment (Benestar) [inline src/Statement/StatementList.php]: I fixed this directly in master as this change does not need any discussion and makes all the tests fail.

Comment (thiemowmde) [inline src/Statement/StatementList.php]: Sure. Thanks for documenting it.

**Pull request merged**